infrastructure/modules/default-images/outputs.tf

```diff
@@ -0,0 +1,27 @@
+output "armonik_versions" {
+  description = "Versions of all the ArmoniK components"
+  value       = var.armonik_versions
+}
+
+output "image_tags" {
+  description = "Tags of images used"
+  value = merge({
+    for image in var.armonik_images.infra :
+    image => var.armonik_versions.infra
+    }, {
+    for image in var.armonik_images.core :
+    image => var.armonik_versions.core
+    }, {
+    for image in var.armonik_images.api :
+    image => var.armonik_versions.api
+    }, {
+    for image in var.armonik_images.gui :
+    image => var.armonik_versions.gui
+    }, {
+    for image in var.armonik_images.extcsharp :
+    image => var.armonik_versions.extcsharp
+    }, {
+    for image in var.armonik_images.samples :
+    image => var.armonik_versions.samples
+  }, var.image_tags)
+}
```

Comment:
Preferably use a map "registry:tag" for all images (those of ArmoniK components and others) and don't split into 2 outputs.

Reply:
I'm not sure about your "registry:tag" comment, but will try to answer nonetheless.

If you want to have a set of "registry:tag" string instead of a map "registry" => "key", there is multiple reasons why I have set it up like this:
- The output is a dictionary that enable to look up the default tag for a given image. This cannot be changed to a set as we do need the ability to look up.
- All the ArmoniK modules require the image name and the tag to be split into two arguments.

If you meant that this output should also include the ArmoniK images, it is already the case. This is the purpose of the merge. I have put another output `armonik_versions` with ArmoniK versions only for reference (for instance, if you want to tag the deployment). The purpose of `armonik_versions` is not to find image tags, but to find repo tags.